WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -252,7 +267,7 @@ public async Task<InputRegistrationResponse> RegisterInputAsync(InputRegistratio
 			var coin = await InputRegistrationHandler.OutpointToCoinAsync(request, Prison, Rpc, Config).ConfigureAwait(false);
 			using (await AsyncLock.LockAsync().ConfigureAwait(false))
 			{
-				var registeredCoins = Rounds.SelectMany(r => r.Alices.Select(a => a.Coin));
+				var registeredCoins = Rounds.Where(x => !(x.Phase == Phase.Ended && !x.IsTransactionBroadcasted)).SelectMany(r => r.Alices.Select(a => a.Coin));
```

Comment:
no it can't, because this is negated, we also want inactive rounds that succeeded, those coins should be considered register even if the round is no longer active

Reply:
It seems i don't understand the logic. This predicate

```c#
Where(x => !(x.Phase == Phase.Ended && !x.IsTransactionBroadcasted)
```

Is equivalent to:
```c#
Where(x => x.Phase != Phase.Ended || x.IsTransactionBroadcasted)
```

I understand the part on the left of the `or` operator but why is the part on the right correct?

Reply:
if the phase is not ended, or it has ended but the transaction has been broadcasted, then the coins registered in that round should not be allowed into a new round (they should only be allowed into blame rounds)

Reply:
I see. Well, in that case the the second style is better.

Reply:
if it's ended and broadcast then trying to use the same coin in another round will result in a double spend, so that should not be allowed (input registration will disallow it when checking the tx output.

this code has since been changed in #6065 and this duplicate finding logic has been removed so i would prefer to just leave it as is, one less rebase for me...